src/full/Agda/Compiler/Common.hs

```diff
@@ -168,15 +169,18 @@ inCompilerEnv checkResult cont = do
     -- Unfortunately, a pragma option is stored in the interface file as
     -- just a list of strings, thus, the solution is a bit of hack:
     -- We match on whether @["--no-main"]@ is one of the stored options.
-    when (["--no-main"] `elem` iFilePragmaOptions mainI) $
+    when (["--no-main"] `elem`
+          map pragmaStrings (iFilePragmaOptions mainI)) $
```

Comment:
Could you factor out `map pragmaStrings (iFilePragmaOptions mainI)`?  It appears several times subsequently and giving it a name would make the code more succinct and more readable.

Reply:
OK.